### PR TITLE
feat(dependency): Upgrade go version to 1.14

### PR DESCRIPTION
**Pull request by harshthakur9030**
## Pull Request template

**Why is this PR required? What issue does it fix?**:
This PR upgrades to go version to 1.14

**What this PR does?**:
It makes the CI and the repo's go version consistent with each other.

**Does this PR require any upgrade changes?**:
Yes, go version is upgraded

**If the changes in this PR are manually verified, list down the scenarios covered:**:
Manually built the binary locally

**Any additional information for your reviewer?** : 
Should be merged after removing gox PR as the go.mod file doesn't contain gox as a dependency anymore. 
https://github.com/openebs/node-disk-manager/pull/456

**Checklist:**
- [ ] Fixes #<issue number>
- [x] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Has the change log section been updated? 
- [ ] Commit has unit tests
- [ ] Commit has integration tests
- [ ] (Optional) Are upgrade changes included in this PR? If not, mention the issue/PR to track: 
- [ ] (Optional) If documentation changes are required, which issue on https://github.com/openebs/openebs-docs is used to track them:

### CODE REVIEW

File: vendor/golang.org/x/sync/semaphore/semaphore.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
+// Copyright 2017 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package semaphore provides a weighted semaphore implementation.
+package semaphore // import "golang.org/x/sync/semaphore"
```
Comment (akhilerm) [inline vendor/golang.org/x/sync/semaphore/semaphore.go:6]: question: wasn't this included as part of the Rescan implementation.?

File: vendor/modules.txt [hunk 2]
```diff
@@ -1,10 +1,12 @@
 # github.com/appscode/jsonpatch v0.0.0-20190108182946-7c0e3b262f30
+## explicit
```
Comment (akhilerm) [inline vendor/modules.txt:2]: question: why some modules are marked as `explicit` with version upgrade?

File: vendor/golang.org/x/sync/semaphore/semaphore.go (hunk 1 above)
Comment (harshthakur9030) [reply vendor/golang.org/x/sync/semaphore/semaphore.go:6]: Yes. I am not sure why upgrading go version caused the change in vendor files related to this package.

**Comment by akhilerm** [comment]
@harshthakur9030 Can you also add a changelog for this version upgrade?

File: vendor/modules.txt (hunk 2 above)
Comment (harshthakur9030) [reply vendor/modules.txt:2]: Section about vendoring explains that: https://golang.org/doc/go1.14

File: changelogs/unreleased/459-harshthakur9030 [hunk 3]
```diff
@@ -0,0 +1 @@
+Upgrade go version to 1.14 to maintain consistency across repo and CI.
```
Comment (akhilerm) [inline changelogs/unreleased/459-harshthakur9030:1]: A small correction in the changelog message.
```suggestion
Upgrade go version to 1.14
```

**Pull request merged**